### PR TITLE
feat(JOML): migrate PocketStructreDefinition

### PR DESCRIPTION
**Pull request by pollend**
I replaced Transform with a JOML matrix but I'm not sure about the implementation.

### CODE REVIEW

File: src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java [hunk 1]
```diff
@@ -113,18 +115,19 @@ public DiffusePocketStructure(Transform transform, Random random, Vector3ic chun
             if (rMax < 0) {
                 rMax = 0;
             }
-            float[] bb = new float[]{-rMax, -rMax, -rMax, rMax, rMax, rMax};
-            transform.transformBB(bb);
+            Vector3f min = new Vector3f();
+            Vector3f max = new Vector3f();
+            transform.frustumAabb(min, max);
```
Comment (4Denthusiast) [inline src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java]: You need to take `rMax` into account too. This computes the bounding box for a 2*2*2 cube. You could do this by adding `transform.scale(rMax);` before `transform.frustumAabb(min, max);` (or something that has the same effect on `min` and `max` without messing up `transform`).

File: src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java [hunk 2]
```diff
@@ -113,18 +115,19 @@ public DiffusePocketStructure(Transform transform, Random random, Vector3ic chun
...
 
-            float minX = Math.min(bb[0], bb[3]);
-            float minY = Math.min(bb[1], bb[4]);
-            float minZ = Math.min(bb[2], bb[5]);
+            float minX = Math.min(min.x, max.x);
+            float minY = Math.min(min.y, max.y);
+            float minZ = Math.min(min.z, max.z);
```
Comment (4Denthusiast) [inline src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java]: The `Math.min`s (and `Math.max`s later) are unnecessary. The coordinates of `min` are already <= the coordinates of `max`.

File: src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java [hunk 3]
```diff
@@ -133,9 +136,9 @@ public DiffusePocketStructure(Transform transform, Random random, Vector3ic chun
             noiseLevels = (maxSize <= 1) ? 0 : (int) (java.lang.Math.log(maxSize) / java.lang.Math.log(2) + 0.5F);
 
             // store transforms
-            mat = transform.clone();
+            mat = new Matrix4f(transform);
             if (transform.determinant() != 0) {
-                invMat = transform.inverse();  // note - this alters the transform argument
+                invMat = transform.invert(new Matrix4f());  // note - this alters the transform argument
```
Comment (4Denthusiast) [inline src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java]: This does not actually alter the `transform` argument. This is fine, but the comment should be removed.

File: src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java [hunk 4]
```diff
@@ -176,12 +179,12 @@ public void generateStructure(StructureCallback callback) {
...
 
                         // transform into local coordinates
-                        pos[0] = x + 0.5F;
-                        pos[1] = y + 0.5F;
-                        pos[2] = z + 0.5F;
-                        invMat.transformVector(pos);
+                        pos.x = x + 0.5F;
+                        pos.y = y + 0.5F;
+                        pos.z = z + 0.5F;
+                        invMat.transformDirection(pos);
```
Comment (4Denthusiast) [inline src/main/java/org/terasology/customOreGen/PocketStructureDefinition.java]: This should be just `transform` (or equivalently, `pos.mul(invMat)`), not `transformDirection`. The translation component is important too.

**Review by 4Denthusiast — DISMISSED** [review]

**Comment by jdrueckert** [comment]
@pollend can you take another look at this pls, so we can get it ready and merged?

**Review by skaldarnar — APPROVED** [review]

**Pull request merged**